### PR TITLE
Feature/plcn 67 command and handler for client entity

### PR DESCRIPTION
**Pull request by AntonEsbjergIT-Minds**

### CODE REVIEW

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+
+		Client Client = new(Guid.NewGuid());
+		Client ClientHubSpot = new(Guid.NewGuid());
+		ClientContact clientContactHubSpot = new(Guid.NewGuid());
+		ClientHubSpot.ClientContacts.Add(clientContactHubSpot);
+		MockSetupForNumAssociatedContactsCaseSuccess(Client, ClientHubSpot);
+		// Act
+		Result result = await _uut.Handle(command, _cancellationToken);
+		//Assert
+		Assert.Empty(Client.ClientContacts);
```
Comment (Nicolai-Bak) [inline Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:690]: If the hubspotclient has clientcontacts, why should this be empty?

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 2]
```diff
@@ -0,0 +1,810 @@
...
+			HubSpotContactId = Guid.NewGuid().ToString()
+		};
+		ClientHubSpot.ClientContacts.Add(clientContactHubSpot);
+		MockSetupForNumAssociatedContactsCaseSuccess(client, ClientHubSpot);
+
+		// Act
+		Result result = await _uut.Handle(command, _cancellationToken);
+		//Assert
+		Assert.False(client.ClientContacts.First().IsActive);
+		_unitOfWorkMock.Verify(x => x.ClientRepository.Update(client));
```
Comment (Nicolai-Bak) [inline Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:728]: add `, Times.Once()` on all verify update

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 3]
```diff
@@ -0,0 +1,810 @@
...
+		//Assert
+		Assert.Empty(Client.ClientContacts);
+		_unitOfWorkMock.Verify(x => x.ClientRepository.Update(Client));
+		_unitOfWorkMock.Verify(x => x.SaveAsync(_cancellationToken), Times.Once());
+		Assert.True(result.IsSuccess);
+		Assert.Equal(Error.None, result.Error);
+	}
+	[Theory]
+	[InlineData(0, 0, "num_associated_contacts", "0")]
+	public async void Handle_ClientFoundNumAssociatedContactUpdatedButClientContactsInClientNotTheSameAsInHubSpotClient_ReturnsSuccess(
```
Comment (Nicolai-Bak) [inline Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:698]: This and next should not be necessary as hubspot clientid in client and clientHubspot should be the same when both is fetched by id

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 4]
```diff
@@ -0,0 +1,810 @@
...
+
+	[Theory]
+	[InlineData(0, 0, "num_associated_contacts", "0")]
+	public async void Handle_ClientFoundNumAssociatedContactNotUpdatedSupplierNotFound_ReturnsFailure(
+		long objectId,
+		long portalId,
+		string propertyName,
+		string propertyValue)
+	{
+		UpdateClientCommand command = new(objectId, portalId, propertyName, propertyValue);
```
Comment (Nicolai-Bak) [inline Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:458]: Add '// Assert' to tests where it is missing

**Review by Nicolai-Bak — CHANGES_REQUESTED** [review]

File: Pelican.Presentation.Api/Controllers/HubSpotController.cs [hunk 5]
```diff
@@ -36,7 +37,7 @@ public HubSpotController(ISender sender)
 	}
 
 	[HttpPost]
-	[ServiceFilter(typeof(HubSpotValidationFilter))]
+	//[ServiceFilter(typeof(HubSpotValidationFilter))]
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Presentation.Api/Controllers/HubSpotController.cs:40]: Removed Comment

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandValidatorUnitTest.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+namespace Pelican.Application.Test.Clients.Commands.UpdateClient;
+public class UpdateClientCommandValidatorUnitTest
+{
+	private UpdateClientCommandValidator _uut;
+	public UpdateClientCommandValidatorUnitTest()
+	{
+		_uut = new();
+	}
+	[Fact]
+	public void UpdateClientCommandCommandValidator_EmptyOrDefaultInputs_ReturnsError()
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandValidatorUnitTest.cs:14]: Whoops, name has been changed

File: Pelican.Application/Clients/Commands/DeleteClient/DeleteClientCommandHandler.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+using Pelican.Domain.Shared;
+
+namespace Pelican.Application.Clients.Commands.DeleteClient;
+internal sealed class DeleteClientCommandHandler : ICommandHandler<DeleteClientCommand>
+{
+	private readonly IUnitOfWork _unitOfWork;
+
+	public DeleteClientCommandHandler(IUnitOfWork unitOfWork)
+	{
+		_unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(IUnitOfWork));
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/DeleteClient/DeleteClientCommandHandler.cs:13]: Changed it to be nameof parameter

File: Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandValidatorUnitTest.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+public class DeleteClientCommandValidatorUnitTest
+{
+	private readonly DeleteClientCommandValidator _uut;
+	public DeleteClientCommandValidatorUnitTest()
+	{
+		_uut = new();
+	}
+
+	[Fact]
+	public void DeleteClientCommandCommandValidator_EmptyId_ReturnsError()
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandValidatorUnitTest.cs:15]: Whoops, implemented the change

File: Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandValidatorUnitTest.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+		// Assert
+		result.ShouldNotHaveValidationErrorFor(command => command.ObjectId);
+	}
+
+	[Fact]
+	public void DeleteClientCommandCommandValidator_NegativeIdValue_ReturnsNoError()
+	{
+		// Arrange
+		DeleteClientCommand command = new(long.MinValue);
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandValidatorUnitTest.cs:44]: Moved, into one test theory with inline data.

File: Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandHandlerUnitTest.cs [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+using Pelican.Domain.Entities;
+using Pelican.Domain.Shared;
+using Xunit;
+
+namespace Pelican.Application.Test.Clients.Commands.DeleteClient;
+public class DeleteClientCommandHandlerUnitTest
+{
+	private DeleteClientCommandHandler uut;
+	private Mock<IUnitOfWork> fakeUnitOfWork;
+	private CancellationToken cancellation;
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandHandlerUnitTest.cs:13]: Added '_' prefix and also made readonly

File: Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandHandlerUnitTest.cs [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	[Fact]
+	public async void Handle_ClientNotFound_ReturnsSuccess()
+	{
+		//Arrange
+		DeleteClientCommand deleteClientCommand = new(1);
+		fakeUnitOfWork.Setup(x => x
+			.ClientRepository
+			.FindByCondition(It.IsAny<System.Linq.Expressions.Expression<Func<Client, bool>>>()))
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandHandlerUnitTest.cs:41]: Yes. added it and shortened test names.

File: Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandHandlerUnitTest.cs [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+		//Act
+		var result = await uut.Handle(deleteClientCommand, cancellation);
+		//Assert
+		fakeUnitOfWork.Verify(x => x.SaveAsync(cancellation), Times.Never());
+
+		fakeUnitOfWork
+			.Verify(
+				unitOfWork => unitOfWork.ClientRepository
+					.FindByCondition(It.IsAny<System.Linq.Expressions.Expression<Func<Client, bool>>>()),
+			Times.Once());
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandHandlerUnitTest.cs:52]: Added

File: Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandHandlerUnitTest.cs [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+		//Arrange
+		DeleteClientCommand deleteClientCommand = new(1);
+		fakeUnitOfWork.Setup(x => x
+			.ClientRepository
+			.FindByCondition(It.IsAny<System.Linq.Expressions.Expression<Func<Client, bool>>>()))
+			.Returns(Enumerable.Empty<Client>().AsQueryable());
+		//Act
+		var result = await uut.Handle(deleteClientCommand, cancellation);
+		//Assert
+		fakeUnitOfWork.Verify(x => x.SaveAsync(cancellation), Times.Never());
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/DeleteClient/DeleteClientCommandHandlerUnitTest.cs]: Rearranged so saveasync is last

File: Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	private async Task<Result> UpdateClientContacts(Client localClient, long portalId, long objectId, string propertyName, CancellationToken cancellationToken)
+	{
+		Supplier? supplier = _unitOfWork
+				.SupplierRepository
+				.FindByCondition(supplier => supplier.HubSpotId == portalId)
+				.FirstOrDefault();
+
+		if (supplier is null || supplier.RefreshToken is null || supplier.RefreshToken == "")
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs:109]: Done :D

File: Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	private async Task<Result> GetClientFromHubSpot(long portalId, long objectId, CancellationToken cancellationToken)
+	{
+		Supplier? supplier = _unitOfWork
+				.SupplierRepository
+				.FindByCondition(supplier => supplier.HubSpotId == portalId)
+				.FirstOrDefault();
+
+		if (supplier is null || supplier.RefreshToken is null || supplier.RefreshToken == "")
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs:76]: Done :D

File: Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+		{
+			return Result.Failure<Client>(Error.NullValue);
+		}
+
+		Result<string> accessTokenResult = await _hubSpotAuthorizationService
+			.RefreshAccessTokenAsync(supplier.RefreshToken, cancellationToken);
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs:115]: Created a new method to get access token

File: Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+		foreach (var item in localClient.ClientContacts)
+		{
+			if (!result.Value.ClientContacts.Any(c => c.HubSpotClientId == item.HubSpotClientId && c.HubSpotContactId == item.HubSpotContactId))
+			{
+				item.IsActive = false;
+			}
+		}
+		_unitOfWork
+					.ClientRepository
+					.Update(localClient);
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs:137]: Done

File: Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+		if (result.IsFailure)
+		{
+			return Result.Failure<Client>(result.Error);
+		}
+		foreach (var item in localClient.ClientContacts)
+		{
+			if (!result.Value.ClientContacts.Any(c => c.HubSpotClientId == item.HubSpotClientId && c.HubSpotContactId == item.HubSpotContactId))
+			{
+				item.IsActive = false;
+			}
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs:133]: Added a new foreach loop that adds any client contacts in result which aren't in local to local

File: Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+
+		Result<string> accessTokenResult = await _hubSpotAuthorizationService
+			.RefreshAccessTokenAsync(supplier.RefreshToken, cancellationToken);
+
+		if (accessTokenResult.IsFailure)
+		{
+			return Result.Failure<Client>(accessTokenResult.Error);
+		}
+
+		Result<Client> result = await _hubSpotClientService.GetByIdAsync(accessTokenResult.Value, objectId, cancellationToken);
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs:89]: Now adds a contact with contactid.

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 20]
```diff
@@ -0,0 +1,810 @@
...
+using Pelican.Domain.Shared;
+using Xunit;
+
+namespace Pelican.Application.Test.Clients.Commands.UpdateClient;
+
+public class UpdateClientCommandHandlerTests
+{
+	private readonly UpdateClientCommandHandler _uut;
+	private readonly Mock<IUnitOfWork> _unitOfWorkMock;
+	private readonly Mock<IGenericRepository<Client>> _ClientRepositoryMock;
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:16]: Changed the name

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 21]
```diff
@@ -0,0 +1,810 @@
...
+				.FindByCondition(It.IsAny<System.Linq.Expressions.Expression<Func<Supplier, bool>>>()))
+				.Returns(Enumerable.Empty<Supplier>().AsQueryable());
+
+		// Act
+		Result result = await _uut.Handle(command, _cancellationToken);
+
+		// Assert
+		_unitOfWorkMock.Verify(unitOfWork => unitOfWork.ClientRepository
+				.FindByCondition(Client => Client.HubSpotId == command.ObjectId.ToString()),
+				Times.Once());
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:76]: Tried to fix indentation

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 22]
```diff
@@ -0,0 +1,810 @@
...
+		// Arrange
+		UpdateClientCommand command = new(objectId, portalId, propertyName, propertyValue);
+
+		Supplier supplier = new(Guid.NewGuid())
+		{
+			RefreshToken = "",
+		};
+
+		_unitOfWorkMock.Setup(unitOfWork => unitOfWork.ClientRepository
+				.FindByCondition(It.IsAny<System.Linq.Expressions.Expression<Func<Client, bool>>>()))
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs]: Done

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 23]
```diff
@@ -0,0 +1,810 @@
...
+		_unitOfWorkMock.Setup(unitOfWork => unitOfWork
+			.SupplierRepository
+			.FindByCondition(It.IsAny<System.Linq.Expressions.Expression<Func<Supplier, bool>>>()))
+			.Returns(new List<Supplier> { supplier }.AsQueryable());
+
+		// Act
+		Result result = await _uut.Handle(command, _cancellationToken);
+
+		// Assert
+		_unitOfWorkMock.Verify(x => x.SupplierRepository.FindByCondition(s => s.HubSpotId == portalId));
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:161]: I concluded that, it was a bit superflous, I just checked it the first time.

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 24]
```diff
@@ -0,0 +1,810 @@
...
+			.ReturnsAsync(Result.Success("token"));
+
+		_hubSpotClientServiceMock.Setup(service => service
+			.GetByIdAsync(It.IsAny<string>(), It.IsAny<long>(), It.IsAny<CancellationToken>()))
+			.ReturnsAsync(Client);
+
+		// Act
+		Result result = await _uut.Handle(command, _cancellationToken);
+
+		// Assert
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:276]: Added Verify on Createasync()

File: Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs [hunk 25]
```diff
@@ -0,0 +1,141 @@
...
+				break;
+			case "industry":
+				client.Segment = command.PropertyValue;
+				break;
+			case "website":
+				client.Website = command.PropertyValue;
+				break;
+			case "num_associated_contacts":
+				return await UpdateClientContacts(client, command.PortalId, command.ObjectId, command.PropertyName, cancellationToken);
+			default:
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs:56]: Added exception

[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application/Clients/Commands/UpdateClient/UpdateClientCommandHandler.cs:56]: And a test for this scenario

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 26]
```diff
@@ -0,0 +1,810 @@
...
+	public async void Handle_ClientFoundButPropertyNameDoesNotMatchAnyCase_ReturnsSuccess(
+		long objectId,
+		long portalId,
+		string propertyName,
+		string propertyValue)
+	{
+		// Arrange
+		UpdateClientCommand command = new(objectId, portalId, propertyName, propertyValue);
+
+		Client Client = new(Guid.NewGuid());
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:292]: Changed

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 27]
```diff
@@ -0,0 +1,810 @@
...
+
+		Client updatedClient = Client;
+		updatedClient.Name = command.PropertyValue;
```
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:327]: Changed it to Assert.Equal(command.PropertyValue, client.RelevanctProperty)

Relevant Property is in this case .Name

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs (hunk 4 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:459]: Added //Arrange to test where it was missing

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs (hunk 1 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:691]: With the changes you suggested earlier you are right, I have changed the test now.

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs (hunk 3 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:699]: Removed

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs (hunk 2 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:729]: Whoops, have been added

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 28]
```diff
@@ -0,0 +1,1057 @@
...
+
+		_uut = new(
+			_unitOfWorkMock.Object,
+			_hubSpotClientServiceMock.Object,
+			_hubSpotAuthorizationServiceMock.Object);
+
+		_cancellationToken = new();
+	}
+
+	private void SetupUnitOfWork()
```
Comment (llh-itm) [inline Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:38]: this method is only called in the test below it but is immediately overwritten in the test anyway - should be removed

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs [hunk 29]
```diff
@@ -0,0 +1,1057 @@
...
+	public async void Handle_ClientNotFoundSupplierNotFound_ReturnsFailureErrorNullValue(
+		long objectId,
+		long portalId,
+		string propertyName,
+		string propertyValue)
+	{
+		// Arrange
+		UpdateClientCommand command = new(objectId, portalId, propertyName, propertyValue);
+
+		SetupUnitOfWork();
```
Comment (llh-itm) [inline Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:60]: does nothing, remove

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs (hunk 28 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:38]: Has been removed

File: Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs (hunk 29 above)
Comment (AntonEsbjergIT-Minds) [reply Pelican.Application.Test/Clients/Commands/UpdateClient/UpdateClientCommandHandlerUnitTest.cs:60]: Removed

**Review by Nicolai-Bak — APPROVED** [review]

**Pull request merged**